website/docs/ReactNavigation.md

```diff
@@ -198,6 +202,155 @@ describe('Testing react navigation', () => {
+
+### Setting up
+
+Install the packages required for React Navigation. For this example, we will use a [drawer navigator](https://reactnavigation.org/docs/drawer-navigator/) to transition between a home screen and an additional screen.
+
+```
+$ yarn add @react-native-community/masked-view @react-navigation/native @react-navigation/drawer react-native-gesture-handler react-native-reanimated react-native-safe-area-context react-native-screens
+```
+
+Create a [`./DrawerAppNavigator.js`](https://github.com/callstack/react-native-testing-library/blob/master/examples/reactnavigation/src/DrawerAppNavigator.js) component which will list the navigation stack:
+
+```jsx
+import 'react-native-gesture-handler';
+import React from 'react';
+import {createDrawerNavigator} from '@react-navigation/drawer';
+
+const { Screen, Navigator } = createDrawerNavigator();
+
+export default function Navigation() {
+  const options = {};
+
+  return (
+    <Navigator>
+      <Screen name="Home" component={HomeScreen} />
+      <Screen name="Notifications" component={NotificationsScreen} />
+    </Navigator>
+  );
+}
+```
+
+Create your two screens which we will transition to and from:
+
+```jsx
+function HomeScreen({ navigation }) {
+  return (
+    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
+      <Text>Welcome!</Text>
+      <Button
+        onPress={() => navigation.navigate('Notifications')}
+        title="Go to notifications"
+      />
+    </View>
+  );
+}
+
+function NotificationsScreen({ navigation }) {
+  return (
+    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
+      <Text>This is the notifications screen</Text>
+      <Button onPress={() => navigation.goBack()} title="Go back home" />
+    </View>
+  );
+}
+```
+
+### Setting up the test environment
+
+Install required dev dependencies:
+
+```
+$ yarn add -D jest @testing-library/react-native
+```
+
+Create a [`mock file`](https://github.com/callstack/react-native-testing-library/blob/master/examples/reactnavigation/jest-mocks.js) necessary for your tests:
+
+```jsx
+import 'react-native-gesture-handler/jestSetup';
+
+jest.mock('react-native-reanimated', () => {
+  const Reanimated = require('react-native-reanimated/mock');
+
+  // The mock for `call` immediately calls the callback which is incorrect
+  // So we override it with a no-op
+  Reanimated.default.call = () => {};
+
+  return Reanimated;
+});
+
+// Silence the warning: Animated: `useNativeDriver` is not supported because the native animated module is missing
+jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
+```
+
+Create your `jest.config.js` file (or place the following properties in your `package.json` as a "jest" property)
+
+```js
+module.exports = {
+  preset: 'react-native',
+  setupFiles: ['./jest-mocks.js'],
+  transformIgnorePatterns: [
+    'node_modules/(?!(jest-)?react-native|@react-native-community|@react-navigation)',
```

Comment:
```suggestion
    'node_modules/(?!(jest-)?react-native|@react-native-community)',
```

Reply:
Removing the `@react-navigation` here makes the tests fail:

<img width="593" alt="Zrzut ekranu 2020-12-10 o 17 22 07" src="https://user-images.githubusercontent.com/4152181/101799298-4a78a900-3b0c-11eb-9155-e8e39730e594.png">

Reply:
Interesting. I assumed Jest already handled asset files with react-native preset cc @thymikee

Reply:
Jest doesn't transpile node_modules by default. RN preset adjusts the configuration so that only node_modules other than `react-native*` and `@react-native-community*` are not transpiled. `@react-navigation` is not there, so we need to add it manually.

Reply:
@thymikee I'm talking about assets, React Navigation's code doesn't need to be compiled to run in Node.

Reply:
`png` is passed to a custom transformer: https://github.com/facebook/react-native/blob/master/jest-preset.js#L19 and transformers obey "transformIgnorePatterns", so here's why it fails.

Reply:
Could be fixed by using `moduleNameMapper` instead of transformer: https://jestjs.io/docs/en/webpack#handling-static-assets

Reply:
@thymikee do you know why it doesn't use `moduleNameMapper`? probably so people wouldn't have to specify it again when overriding?

Reply:
I don't know why, would have to dig through git blame. Jest handles merging `moduleNameMapper` gracefully, same as transforms